platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/report/generic/PerRootTabRenderer.java

```diff
@@ -200,48 +244,76 @@ private void renderFailure(SerializableFailure failure, SimpleHtmlWriter htmlWri
             }
         }
 
-        private void renderChildren(SimpleHtmlWriter htmlWriter) throws IOException {
-            htmlWriter.startElement("table");
-            htmlWriter.startElement("thead");
-            htmlWriter.startElement("tr");
+        private static String getStatusClass(TestResult.ResultType resultType) {
+            switch (resultType) {
+                case SUCCESS:
+                    return "success";
+                case FAILURE:
+                    return "failures";
+                case SKIPPED:
+                    return "skipped";
+                default:
+                    throw new IllegalStateException();
+            }
+        }
+
+        private static String getFormattedSuccessRate(TestTreeModel.PerRootInfo info) {
+            if (info.getTotalLeafCount() == 0) {
+                return "-";
+            }
+
+            BigDecimal runTests = BigDecimal.valueOf(info.getTotalLeafCount());
+            BigDecimal successful = BigDecimal.valueOf(info.getTotalLeafCount() - info.getFailedLeafCount());
+
+            return successful.divide(runTests, 2, RoundingMode.DOWN).multiply(BigDecimal.valueOf(100)).intValue() + "%";
+        }
 
-            boolean anyNameAndDisplayNameDiffer = Iterables.any(
-                getCurrentModel().getChildrenOf(rootIndex),
-                childModel -> Iterables.any(
-                    childModel.getPerRootInfo().get(rootIndex),
+        private static final class ChildListRenderer extends ReportRenderer<TestTreeModel, SimpleHtmlWriter> {
```

Comment:
💅 Maybe `TestResultsTableRenderer`?  I think if someone came to this file looking ot adjust the table, `ChildListRenderer` is not obvious.  

Also, in the context of this class alone, "child" and "children" aren't the clearest.  It's just rendering a table of test results - it doesn't know or care that they are in any sort of hierarchical relationship with anything.